src/query/sql/src/async_function/sequence_async_function.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use databend_common_ast::ast::ColumnID;
+use databend_common_ast::ast::Expr;
+use databend_common_catalog::catalog::Catalog;
+use databend_common_exception::ErrorCode;
+use databend_common_exception::Result;
+use databend_common_exception::Span;
+use databend_common_expression::types::DataType;
+use databend_common_expression::types::NumberDataType;
+use databend_common_expression::types::NumberScalar;
+use databend_common_expression::Scalar;
+use databend_common_meta_app::schema::GetSequenceNextValueReq;
+use databend_common_meta_app::schema::GetSequenceReq;
+use databend_common_meta_app::schema::SequenceIdent;
+use databend_common_meta_app::tenant::Tenant;
+
+use crate::plans::AsyncFunctionCall;
+use crate::plans::ConstantExpr;
+use crate::AsyncFunction;
+use crate::ScalarExpr;
+
+pub struct SequenceAsyncFunction {
+    func_name: String,
+    return_type: DataType,
+}
+
+impl SequenceAsyncFunction {
+    pub fn create() -> Arc<dyn AsyncFunction> {
+        Arc::new(SequenceAsyncFunction {
+            func_name: "nextval".to_string(),
+            return_type: DataType::Number(NumberDataType::UInt64),
+        })
+    }
+}
+
+#[async_trait::async_trait]
+impl AsyncFunction for SequenceAsyncFunction {
+    fn func_name(&self) -> String {
+        self.func_name.clone()
+    }
+
+    async fn resolve(
+        &self,
+        span: Span,
+        tenant: Tenant,
+        catalog: Arc<dyn Catalog>,
+        arguments: &[Expr],
+    ) -> Result<Option<Box<(ScalarExpr, DataType)>>> {
+        if arguments.len() != 1 {
+            return Err(ErrorCode::SemanticError(
+                "nextval function need one argument".to_string(),
```

Comment:
better to add the actual number of arguments in the error message, like add `but got ...`.